src/rhsmlib/facts/hwprobe.py

```diff
@@ -565,6 +565,40 @@ def get_ls_cpu_info(self):
 
         return lscpu_info
 
+    def _get_ipv4_addr_list(self):
+        """
+        When DNS record is not configured properly for the system, then try to
+        get list of all IPv4 addresses from all devices. Return 127.0.0.1 only
+        in situation when there is only loopback device.
+        :return: list of IPv4 addresses
+        """
+        addr_list = []
+        interface_info = ethtool.get_interfaces_info(ethtool.get_devices())
+        for info in interface_info:
+            for addr in info.get_ipv4_addresses():
```

Comment:
Could we just do

```python
addr_list = [addr for addr in info.get_ipv4_addresses() if addr.address != '127.0.0.1']
```

Reply:
I prefer list comprehension too, but only for simple situation. There are two for loops. I can rewrite it using list comprehension:

```python
addr_list = [addr.address for info in interface_info for addr in info.get_ipv4_addresses() if addr.address != '127.0.0.1']
```

but I consider original code more clear and readable ;-).

Reply:
Okay.  Looks good to me otherwise.  Those Jenkins failures are false positives I believe.